File: adapter/java/concrete/VlcPlayer.java
package adapter.java.concrete;

import adapter.java.legacy.AdvancedMediaPlayer;

public class VlcPlayer implements AdvancedMediaPlayer{
    @Override
    public void playVlc(String fileName) {
        System.out.println("Reproduciendo archivo vlc. Nombre: " + fileName);
    }

    @Override
    public void playMp4(String fileName) {
        // No hace nada
    }
}

File: adapter/php/concrete/VlcPlayer.php
<?php
namespace Concrete;

require_once 'legacy/AdvancedMediaPlayer.php';
use Legacy\AdvancedMediaPlayer;

// Implementacion concreta de Adaptee
class VlcPlayer implements AdvancedMediaPlayer {
    public function playVlc($fileName) {
        echo "Reproduciendo archivo vlc. Nombre: " . $fileName . "\n";
    }
    public function playMp4($fileName) {
        // no hace nada
    }
}